Assets/Scripts/Robot/CommunicationManager.cs

```diff
@@ -378,11 +378,9 @@ public List<SensedObject<int>> SenseNearbyRobots(int id)
                 {
                     continue;
                 }
-
                 var comInfo = _adjacencyMatrix![(id, robot.id)];
-                if ((comInfo.Distance > _robotConstraints.SenseNearbyAgentsRange && !_robotConstraints.MaterialCommunication) ||
-                   (comInfo.WallCellsDistance > 0 && _robotConstraints.SenseNearbyAgentsBlockedByWalls) ||
-                   (!comInfo.TransmissionSuccessful && _robotConstraints.MaterialCommunication))
+
+                if (!comInfo.TransmissionSuccessful || comInfo.Distance > _robotConstraints.SenseNearbyAgentsRange)
```

Comment:
This should NOT be changed!

Reply:
This is not even explained in the PR

Reply:
This is due to our improved raytracing, we do not need all the other checks now. If we dont have a TransmissionSuccessful we do not have the ability to communicate.

Reply:
This fucks with sensing though. Sensing is not the same as "inside communication range"